pkg/htmx/gin.go

```diff
@@ -55,12 +58,24 @@ func GetHxCurrentPath(ctx *gin.Context) string {
 }
 
 func HasMatchingParentPath(ctx *gin.Context) bool {
-	currentUrlParts := strings.Split(GetHxCurrentPath(ctx), "/")
-	shortCurrent := path.Join(currentUrlParts[0 : len(currentUrlParts)-1]...)
-	requestUrlParts := strings.Split(ctx.Request.URL.Path, "/")
-	shortRequest := path.Join(requestUrlParts[0 : len(requestUrlParts)-1]...)
-	fmt.Println(shortCurrent, shortRequest)
-	return shortCurrent == shortRequest
+	currentPath := GetHxCurrentPath(ctx)
+	currentPathParts := strings.Split(currentPath, "/")
+
+	requestPathParts := strings.Split(ctx.Request.URL.Path, "/")
+	requestPathParent := path.Join(requestPathParts[0 : len(requestPathParts)-1]...)
+
+	// If the currentPath is the parent, then check that against the requestedPath's parent
+	if len(currentPathParts) == len(requestPathParts)-1 {
+		return strings.TrimPrefix(currentPath, "/") == requestPathParent
+	}
+
+	// Else, check the equality of parent paths
+	currentPathParent := path.Join(currentPathParts[0 : len(currentPathParts)-1]...)
+	return currentPathParent == requestPathParent
```

Comment:
This function was created as a check for needing to change the active sidenav link to know if the handler must return a newly rendered sidenav. However, when navigating to a parent path, it was failing this check and thus would render a new sidenav that included the same active link. This function will now also check to see if the current path is the parent of the requested path, meaning they will have the same active sidenav link and can skip rendering.

Reply:
Through some quick testing, also noticed that when navigating from a child to the parent list page, it was rendering the sidenav because it wasn't checking for matching parents. I updated the logic again in this function to include a check for when the request path is a potential parent of the current path and updated the respective handlers to check with it to potentially skip rendering the sidenav when possible.